api/src/routing.js

```diff
@@ -22,12 +22,14 @@ const _ = require('underscore'),
       upgrade = require('./controllers/upgrade'),
       settings = require('./controllers/settings'),
       bulkDocs = require('./controllers/bulk-docs'),
+      authorizationMiddleware = require('./middleware/authorization'),
       createUserDb = require('./controllers/create-user-db'),
       createDomain = require('domain').create,
       staticResources = /\/(templates|static)\//,
       favicon = /\/icon_\d+.ico$/,
-      pathPrefix = '/' + db.settings.db + '/',
-      appPrefix = pathPrefix + '_design/' + db.settings.ddoc + '/_rewrite/',
+      pathPrefix = '\/+' + db.settings.db + '\/+',
+      appPathPrefix = '/' + db.settings.db + '/',
+      appPrefix = appPathPrefix + '_design/' + db.settings.ddoc + '/_rewrite/',
```

Comment:
`pathPrefix` is used for Express routes to capture irregular URL's like `//medic//_changes` which are valid and are resolved by CouchDB, while `appPathPrefix` is used for redirecting and such.